tests/integration/schema/updates/add/field/kind/foreign_object_array_test.go

```diff
@@ -954,3 +928,79 @@ func TestSchemaUpdatesAddFieldKindForeignObjectArray_SecondaryObjectKindAndSchem
 	}
 	testUtils.ExecuteTestCase(t, test)
 }
+
+func TestSchemaUpdatesAddFieldKindForeignObjectArray_MissingPrimaryIDField(t *testing.T) {
+	key1 := "bae-decf6467-4c7c-50d7-b09d-0a7097ef6bad"
+
+	test := testUtils.TestCase{
+		Description: "Test schema update, add field with kind foreign object array (17), with auto id field generation",
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						name: String
+					}
+				`,
```

Comment:
suggestion: Although it is interesting to show that this is possible, it would be nice to have a test also for two separate schemas. It would help to show that the `relation_id` field is only set on the primary.

Reply:
> Although it is interesting to show that this is possible, it would be nice to have a test also for two separate schemas

I'm not sure I follow what makes this interesting, it is just a normal one-many relationship and adding a second schema to the test just complicates things.

> It would help to show that the relation_id field is only set on the primary.

I can't see how that would help, how do you see it?  I do think it is worth adding a test though to verify that the _id field does not exist on the many side.

- [ ] Assert _id field does not exist on many side.

Reply:
> I'm not sure I follow what makes this interesting, it is just a normal one-many relationship and adding a second schema to the test just complicates things.

It's interesting that we show that we can add self-referencing relationships. But that mean that everything is happening on the one collection. Having two separate collections would ensure that the operation isn't broken when it's not just one collection being manipulated.

Reply:
I see what you mean, but that is not specifc to the changes in this PR (this feature actually is one of the least relevant, as it never cares which object it is hosted on). I'll add a ticket to close the test gap for other related behaviour though.

- [ ] Add test-ticket

Reply:
In my mind it is very related as we need to ensure that the auto-creation is behaving properly for all objects involved.

Reply:
I understand that the auto creation itself is specific to the one object but the patch operation is tested here through an integration test which involves 2 add operations. I feel like it's relevant to test that it works even with two distinct objects.

Reply:
> integration test which involves 2 add operations

This is why it should be done in another ticket.  It should be tested with a simpler use case that isn't cluttered with irrelevancies.

Reply:
It is irrelevant to the changes in this PR.

Reply:
Isn't too different to trying to force a developer working on filters group by to test with a range of groupby order configurations, just because orderby isnt properly tested yet.  OrderBy should be tested, but in a simpler context where extras dont confuse the test.  It also means that if groupBy-filter is removed as a feature, groupBy-orderby is still tested.

And please remember, I am not objecting to the testing of PatchSchema relation definitions for multiple objects (even though it is technically not needed given the implementation).  I am say it should be done in another ticket, as it is not relevant here and will only complicate and dirty both work items.

Reply:
https://github.com/sourcenetwork/defradb/issues/1809

Reply:
I see it as related to the change in this PR. You're using an integration test to test the feature. The integration test doesn't care where the auto-create happens. It just cares that it happens. The most common way of having references is from one object to an other. Not from object to self. I would be quite appropriate to have a test that ensures that the feature works in the most common reference pattern where 2 objects are involved. It would also ensure that a change in the future doesn't result in only one side of the relationship having the relation id auto created.

If you really don't want to do it here, it should probably be done shortly so we don't forget about it.